File: java/src/main/java/mx/iteso/adapter/impl/WordDocument.java
package mx.iteso.adapter.impl;


import mx.iteso.adapter.IWordDocument;
import mx.iteso.adapter.MSLicense;
import mx.iteso.adapter.media.Format;
import mx.iteso.adapter.media.Image;


public class WordDocument implements IWordDocument {
    /**
     * Format.
     */
    private Format format;
    /**
     * Image.
     */
    private Image backgroundImage;
    /**
     * MSLicense.
     */
    private MSLicense msLicense;
    /**
     * msOfficeVersion.
     */
    private float msOfficeVersion;

    /**
     * Class constructor.
     * @param msLicense license number
     */
    public WordDocument(final MSLicense msLicense) {
        this.msLicense = msLicense;
    }

    /**
     * Format getter.
     * @return format
     */
    public Format getFormat() {
        return this.format;
    }

    /**
     * Format setter.
     * @param format new
     */
    public void setFormat(Format format) {
        this.format = format;
    }

    /**
     * Image getter.
     * @return Image
     */
    public Image getBackground() {
        return this.backgroundImage;
    }

    /**
     * msOfficeVersion setter.
     * @param msOfficeVersion number
     */
    public void setMSOfficeVersion(float msOfficeVersion) {
        this.msOfficeVersion = msOfficeVersion;
    }

    /**
     * MSLicense getter.
     * @return
     */
    public MSLicense getLicense() {
        return this.msLicense;
    }

    /**
     * Edir permissions.
     * @param msLicense npi
     * @return
     */
    public boolean restrictEditIfLicenseIsInvalid(final MSLicense msLicense) {
        return false;
    }
}

File: csharp/Adapter/Clases/WordDocument.cs
namespace Adapter.Clases
{
    public class WordDocument : IWordDocument {

        private readonly Font _font;

        private readonly BackgroundImage _backgroundImage;

        private readonly MsLicense _msLicense;

        private float _version;

        private readonly Format _format;

        public WordDocument(Font fnt, MsLicense msLic, Format fmt, BackgroundImage bckimg) {
            _font = fnt;
            _msLicense = msLic;
            _format = fmt;
            _backgroundImage = bckimg;
        }

        public Format GetFormat() {
            return _format;
        }

        public Image GetBackground() {
            return _backgroundImage.GetImage();
        }

        public void SetMsOfficeVersion(float msOfficeVersion) {
            _version = msOfficeVersion;
        }

        public MsLicense GetLicense() {
            return _msLicense;
        }

        public Font GetFont() {
            return _font;
        }

        public float GetVersion() {
            return _version;
        }

        public bool RestrictEditIfLicenseIsInvalid(MsLicense msLic) {
            return _msLicense == msLic;
        }
    }

}